### ISSUE TITLE
Compile error on 32 bits system

### ISSUE THREAD

**Issue by PieterPenninckx** [dpzmick/rust-easyjack#1]
When I try to compile rust-easyjack on a 32 bits system, I get an error that I don't get on a 64 bits system:

error[E0308] in src/client.rs:138:16: expected u32, found u64 (underlined: `opts.bits() as u64`).

**Comment by dpzmick** [comment]
That's not surprising..

You should give this library a try: https://github.com/wmedrano/rust-jack

They are quite a bit farther along than I am. I'm probably going to kill this repo and just use theirs

**Comment by PieterPenninckx** [comment]
In fact I have tried rust-jack. Unfortunately, in rust-jack, the `JackHandler` trait is hard to implement for me in a meaningful way because it is required to be `Send` and the `AudioInPort` does not implement `Send`. Working with a closure is also not an option for me because I want to use a local variable that does not have the `'static` lifetime. ... if I recall correctly. Maybe I should also open an issue for rust-jack.

**Comment by dpzmick** [comment]
That's interesting. I have been just using PortAudio while I work out some
details of my own project. Definitely open an issue with them, but if they
have really funky restrictions on their APIs perhaps I will resume work on
my own wrappers

**Comment by PieterPenninckx** [comment]
I re-tried rust-jack and the API has been changed, so my previous comments on the rust-jack API are outdated. Despite the improvements, I wasn't able implement the `JackHandler` trait, so I opened an issue (rust-jack issue #60).

I am preparing a fix to make rust-easyjack compile on 32 bits systems (it's really simple (the fix, I mean)); I only have to test if it still works on a 64 bits system.
